### PR TITLE
List Audit Events Implementation

### PR DESCRIPTION
**Pull request by ramrutha497**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : Yes
- I have compiled the code successfully my laptop : Yes
- I have written examples to demonstrates the usage of the newly implemented APIs : Yes
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : Yes

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/ApiAuditEventLogs.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by ramrutha on 13/02/2023.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiAuditEventLogs {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ApiAuditEventLogs.java:15]: Rename to ApiAuditLogEvents.
Move this file to C:\Github\spotinst-sdk-java\src\main\java\com\spotinst\sdkjava\model\api\admin\account

File: src/main/java/com/spotinst/sdkjava/model/AuditEventLogs.java [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by ramrutha on 13/02/2023.
+ */
+public class AuditEventLogs {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/AuditEventLogs.java:11]: Rename to AuditLogEvents
Move to C:\Github\spotinst-sdk-java\src\main\java\com\spotinst\sdkjava\model\bl\admin\account

File: src/main/java/com/spotinst/sdkjava/model/service/admin/account/SpotAccountAdminService.java [hunk 3]
```diff
@@ -153,4 +154,51 @@ public static Boolean deleteAccount(String accountId, String authToken) throws S
         return retVal;
     }
 
+    public static List<ApiAuditEventLogs> listAuditEventLogs(String authToken, String accountId, String fromDate, String responseStatus, String toDate) throws SpotinstHttpException {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/service/admin/account/SpotAccountAdminService.java:157]: move responseStatus as last parameter.

File: src/main/java/com/spotinst/sdkjava/model/repo/admin/account/SpotAccountAdminRepo.java [hunk 4]
```diff
@@ -90,4 +92,22 @@ public RepoGenericResponse<Boolean> delete(String identifier, String authToken,
         return retVal;
     }
 
+    @Override
+    // The base 'IRepository' class getAll() method requires 3 parameters but this implementation needs only
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/repo/admin/account/SpotAccountAdminRepo.java:96]: Make responseStatus as last parameter.

Comment (anuragsharma-123) [reply src/main/java/com/spotinst/sdkjava/model/repo/admin/account/SpotAccountAdminRepo.java:96]: how we are able to use the getAll method from the IRepository?

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
Please review the comments and make fix.

File: src/main/java/com/spotinst/sdkjava/model/ApiAuditEventLogs.java (hunk 1 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/ApiAuditEventLogs.java:15]: Done

File: src/main/java/com/spotinst/sdkjava/model/AuditEventLogs.java (hunk 2 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/AuditEventLogs.java:11]: Done

File: src/main/java/com/spotinst/sdkjava/model/service/admin/account/SpotAccountAdminService.java (hunk 3 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/service/admin/account/SpotAccountAdminService.java:157]: Done

File: src/main/java/com/spotinst/sdkjava/model/repo/admin/account/SpotAccountAdminRepo.java (hunk 4 above)
Comment (ramrutha497) [reply src/main/java/com/spotinst/sdkjava/model/repo/admin/account/SpotAccountAdminRepo.java:96]: Done

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**